src/commands/preload/index.ts

```diff
@@ -161,6 +163,40 @@ Can be repeated to add multiple certificates.\
 			);
 		}
 
+		// Verify that image is not enabled for secure boot. First, confirm it
+		// is a secure boot image with an /opt/*.sig file in the rootA partition.
+		const { explorePartition, BalenaPartition } = await import(
+			'../../utils/image-contents'
+		);
+		const isSecureBoot = await explorePartition<boolean>(
+			params.image,
+			BalenaPartition.ROOTA,
+			async (fs: typeof Fs): Promise<boolean> => {
+				try {
+					const promiseDir = promisify(fs.readdir);
```

Comment:
Maybe instead of using `promisify` here, we can use the native promises api: https://nodejs.org/docs/latest-v20.x/api/fs.html#fspromisesreaddirpath-options to avoid having to do this?

Reply:
@rcooke-warwick this is not the `fs` namespace imported at the top level, but it's the virtual `fs` "namespace" that balena-image-fs gives us.
That being said, since we recently added support for promises in balena-image-fs, @kb2ma should be able to just use v directly.
```suggestion
					const files = await fs.promises.readdir('/opt');
```